lib/unicode.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+const has_icu = process.binding('config').hasIntl;
+const Buffer = require('buffer').Buffer;
+const kDefaultOptions = {};
+
+if (has_icu) {
+  const icu = process.binding('icu');
+
+  const conversions = {
+    'ascii': {
+      'binary': (source) => {
+        return source;
+      },
+      'utf8': (source) => {
+        return source;
+      },
+      'ucs2': (source) => {
+        return icu.convertToUcs2('us-ascii', source);
+      }
+    },
+    'binary': {
+      'ascii': (source) => {
+        return icu.convert('us-ascii', 'iso8859-1', source);
+      },
+      'utf8': (source) => {
+        return icu.convert('utf-8', 'iso8859-1', source);
+      },
+      'ucs2': (source) => {
+        return icu.convertToUcs2('iso8859-1', source);
+      }
+    },
+    'utf8': {
+      'ascii': (source) => {
+        return icu.convert('us-ascii', 'utf-8', source);
+      },
+      'binary': (source) => {
+        return icu.convert('iso-8859-1', 'utf-8', source);
+      },
+      'ucs2': icu.convertToUcs2FromUtf8,
+    },
+    'ucs2': {
+      'ascii': (source) => {
+        if (source.length % 2 !== 0)
+          throw new TypeError('Invalid UCS2 Buffer');
+        return icu.convertFromUcs2('us-ascii', source);
+      },
+      'binary': (source) => {
+        if (source.length % 2 !== 0)
+          throw new TypeError('Invalid UCS2 Buffer');
+        return icu.convertFromUcs2('iso-8859-1', source);
+      },
+      'utf8': (source) => {
+        if (source.length % 2 !== 0)
+          throw new TypeError('Invalid UCS2 Buffer');
+        return icu.convertToUtf8FromUcs2(source);
+      }
+    }
+  };
+
+  exports.transcode = function transcode(source, from_enc, to_enc, options) {
+    if (!source) return;
```

Comment:
Why the source is not defined, it just returns undefined, but `source.buffer` is not an `ArrayBuffer`, the program throws a `TypeError`, use `throw` instead of `return` might be reasonable here?